src/ParticleDA.jl

```diff
@@ -42,6 +42,12 @@ Return the number of state variables.
 """
 function get_n_state_var end
 
+"""
+    ParticleDA.get_indices(model_data) -> Vector{Int}
+
+Return the vector containing the indices of assimilated values in the state vector
+"""
+function get_indices end
```

Comment:
Is the intention here is for this to specifically return the indices along only the 'state variable' axis in to the state array as the method definition for the TDAC model current returns `[1]` which I guess would correspond to the height component? Structuring like this would I think limit us to having the same set of state variables observed at all observation locations / stations (as it seems like this would be specifying the overall set of observation indices as a Cartesian product between the observed state variable indices and indices corresponding to each observation location). While it's probably reasonable to only provide an implementation for that case for now until we need the extra generality, from the perspective of generalising the optimal filter implementation I would say there is a benefit to trying to avoid depending on the state particles being of a particular spatial dimension / shape, and just interfacing with each particle as a flat vector. If `get_indices` instead was specified returned a vector of integer indices in to the flattened states corresponding to observed components that would allow for this abstraction, while still allowing the definitions in particular models to assume that the state variables observed at each location are the same. It might also be worth giving this a more informative name like `get_observed_state_indices`.

Reply:
As suggested I have renamed it to be get_observed_state_indices. I don't see it being too difficult to increase the generality to interface with flattened states. It helps that it is defined on the model side.

Reply:
Another reason for trying to keep the interface in terms of flattened vectors that occurred to me in the meeting yesterday is that this would also make it easier to generalise the particle filter implementation (both bootstrap and optimal proposals) to deal with non-spatial models.